Feat/address completion interface

Dominic-Marcelino opened the pull request.

HimoriTempest reviewed (changes requested):
The extension works as expected but throws a deprecation warning during the building process:
`The legacy JS API is deprecated and will be removed in Dart Sass 2.0.0.`

Addition:
- Pins won't clear after a new address is selected (This doesn't persist through a save or page reload)
![image](https://github.com/user-attachments/assets/a97e251b-6514-4506-aab7-de535b474286)

phazonoverload commented:
And a couple of small creative review feedback pieces: 

- Maybe we can also round the corner on the overall map container, similar to other extensions?
- The Google buttons are a bit different from the search bar. Could we either style the search bar to feel more like the Google inputs, or make the Google inputs feel more 'Directus'. 
- I don’t know if it’s just an illusion, but the “Search Bar” element’s height doesn’t seem to match the height of the “Map” dropdown button.
- And also the text size of the text within the search input field can also be increased by 1 or 2 px since it looks a bit small compared to the others
- The checkbox needs to be inline with the text in the dropdown

That's it 🫡

Dominic-Marcelino commented:
> The extension works as expected but throws a deprecation warning during the building process:
> `The legacy JS API is deprecated and will be removed in Dart Sass 2.0.0.`

This needs to be updatd in the directus extensions sdk. Opened an issue here: https://github.com/directus/directus/issues/24136


>Pins won't clear after a new address is selected (This doesn't persist through a save or page reload)

Fixed in 62747f0b18ecacaad548f2a3ce462bd41fc1328f

>Maybe we can also round the corner on the overall map container, similar to other extensions?

Fixed, now aligned with the map-interface:
![Bildschirmfoto 2024-12-03 um 19 42 45](https://github.com/user-attachments/assets/7019a25f-f138-4a81-8354-0aaeb829fc43)


> - The Google buttons are a bit different from the search bar. Could we either style the search bar to feel more like the Google inputs, or make the Google inputs feel more 'Directus'.
> - I don’t know if it’s just an illusion, but the “Search Bar” element’s height doesn’t seem to match the height of the “Map” dropdown button.
> - And also the text size of the text within the search input field can also be increased by 1 or 2 px since it looks a bit small compared to the others
> - The checkbox needs to be inline with the text in the dropdown

Fixed, replaced all UI-Elements with directus-components:
![Bildschirmfoto 2024-12-04 um 14 00 34](https://github.com/user-attachments/assets/28407ddc-6802-4661-b1fe-22a6fe09ad05)
